### ISSUE TITLE
some questions about heatmaps。

### ISSUE THREAD

**Issue by Sakuralala** [timctho/convolutional-pose-machines-tensorflow#15]
Hey man,I read the paper first but had many questions and then I read your code, that helped a lot. But I still have some questions,can you help me ?
1.By the code(cpm_hand_v2.py:build_loss),I still can't understand how you determined the size of the predicted heatmaps and the ground truth heatmaps.Could you tell me how?
2.By the paper,the loss function is the l2 loss,but in my opinion,each pixel in the predicted heatmap represents an area of the input image,but the ground truth heatmap is just a gaussian kernel and each pixel of the ground truth heatmap just represents one pixel of the input image,so why can we calculate l2 loss between them?
Thanks for your time!

**Comment by timctho** [comment]
1. The network is fully convolutional, output heatmap size is determined by how many down sampling process. In cpm_hand_v2.py, there's only 2 pooling steps related to down sampling, so the output heatmap size is 1/4 of your input image size.

2. You can put gaussian mask of input image size, then resize it into output heatmap size, or just put gaussian mask of output heatmap size. Then you can compute L2 distance by each pixel.

**Comment by Sakuralala** [comment]
Thanks for your replying!I didn't notice that the padding is "same"...And now I probably understand that the pixel of the ground truth heat map also represents an area of the input image. 
Thanks again!
